autospec/tarball.py

```diff
@@ -561,6 +570,11 @@ def process(url_arg, name_arg, ver_arg, target, archives_arg, filemanager):
     # name is created by adding ".gcov" to the package name (if a gcov file
     # exists)
     set_gcov()
+    # build the target path, if necessary
+    if not target and package_dir:
+        target = os.path.join(package_dir, name)
+    # create the path, by cloning existing repo or manually, as necessary
+    call("make clone_%s" % name, check=False, stderr=subprocess.DEVNULL)
```

Comment:
I think autospec should also try to clone the repo with git if the make clone_PKG command fails, since `make clone_PKG` is only defined for current/latest list of packages; packages that formerly existed in the distro may have package repos available, and we want to reuse those.

Then if that second clone attempt fails, autospec can create a new git repo locally, and I believe that's already implemented.

Reply:
Alternatively, we could run `git clone ...` in Makefile.toplevel (ignoring errors) prior to running autospec, and remove this line.

Reply:
We'd need `config.git_pull_uri` in order to call git clone directly, and that's populated in `autospec.package()` right _after_ `tarball.process()`

And of course toplevel git clone will only pick up what's in `packages`, and might take a while, too. I'm not sure what's the right answer here.

Reply:
Oh right, I was confusing the chicken with the egg again :-P

Autospec is detecting the package name, so we cannot run a `git clone` before running `make autospecnew`, since the package name is not yet known. Hmm.

Reply:
Could we do something like the following instead?

```
if package_dir:
    call("git -C {} clone {}".format(package_dir, config.git_pull_uri % {'NAME': tarball.name}))
```

`git_pull_uri` would need to be defined in autospec.conf, of course.

Reply:
But you still don't have config.git_pull_uri at this point.